server/services/prisonerLocationHistoryService.ts

```diff
@@ -0,0 +1,148 @@
+import { format } from 'date-fns'
+import { PrisonApiClient } from '../data/interfaces/prisonApiClient'
+import { Prisoner } from '../interfaces/prisoner'
+import { WhereaboutsApiClient } from '../data/interfaces/whereaboutsApiClient'
+import { CaseNotesApiClient } from '../data/interfaces/caseNotesApiClient'
+import { HistoryForLocationItem } from '../interfaces/prisonApi/historyForLocation'
+import { CellMoveReasonType } from '../interfaces/prisonApi/cellMoveReasonTypes'
+import { CaseLoad } from '../interfaces/caseLoad'
+import { StaffDetails } from '../interfaces/prisonApi/staffDetails'
+import LocationHistoryPageData from '../interfaces/pages/locationHistoryPageData'
+import { formatName, sortByDateTime, putLastNameFirst, hasLength, extractLocation } from '../utils/utils'
+import { RestClientBuilder } from '../data'
+
+export type PrisonerLocationHistoryService = (
+  token: string,
+  prisonerData: Prisoner,
+  agencyId: string,
+  locationId: string,
+  fromDate: string,
+  toDate: string,
+  userCaseLoads: CaseLoad[],
+) => Promise<LocationHistoryPageData>
+
+export default ({
+  prisonApiClientBuilder,
+  whereaboutsApiClientBuilder,
+  caseNotesApiClientBuilder,
+}: {
+  prisonApiClientBuilder: RestClientBuilder<PrisonApiClient>
+  whereaboutsApiClientBuilder: RestClientBuilder<WhereaboutsApiClient>
+  caseNotesApiClientBuilder: RestClientBuilder<CaseNotesApiClient>
+}): PrisonerLocationHistoryService => {
+  const fetchStaffName = (staffId: string, prisonApi: PrisonApiClient) =>
+    prisonApi.getStaffDetails(staffId).then((staff: StaffDetails) => formatName(staff.firstName, '', staff.lastName))
+
+  const fetchWhatHappened = async (
+    bookingId: number,
+    offenderNo: string,
+    bedAssignmentHistorySequence: number,
+    caseNotesApi: CaseNotesApiClient,
+    whereaboutsApi: WhereaboutsApiClient,
+  ) => {
+    try {
+      const cellMoveReason = await whereaboutsApi.getCellMoveReason(bookingId, bedAssignmentHistorySequence)
+      const caseNote = await caseNotesApi.getCaseNote(offenderNo, cellMoveReason.cellMoveReason?.caseNoteId.toString())
+      return caseNote.text
+    } catch (err) {
+      if (err?.status === 404) return null
+      throw err
+    }
```

Comment:
Does this app not have the ignore 404 thing in the client that @davidmiddletonmoj wrote?